mediachain/indexer/mc_simpleclient.py

```diff
@@ -372,19 +373,24 @@ def get_objects(self,
             
             try:
                 started = False
-                with self.transactor.canonical_stream(catchup=catchup_blockchain, timeout=timeout) as stream:
-                    for ref, obj in stream:
+                with self.transactor.canonical_stream(
+                    catchup=catchup_blockchain,
+                    last_known_block_ref=last_known_block_ref,
+                    timeout=timeout
+                ) as stream:
+                    for obj_info in stream:
+                        obj = obj_info['record']
 
                         if (start_id is not False) and (not started):
-                            if obj['data']['_id'] == start_id:
+                            if obj['meta']['data']['_id'] == start_id:
                                 started = True
                             else:
                                 continue
 
-                        yield ref, obj
+                        yield obj_info
```

Comment:
Even better - since this is an API, it'd be nice to have some kind of versioning that's machine-interpretable, and each API caller would do an assert against the version number each time the program ran, prior to actually using the API.

E.g. major number increment indicates reverse-compatibility breaking changes, minor number increment indicates behind-the-scenes enhancements, and each API caller asserts that major number == last major number for which the person manually reviewed the list of breaking changes.

This would help us avoid subtle bugs that automated tests may not catch & help keep all devs keep in sync.